### ISSUE TITLE
implement LoggerSystem on winrt

### ISSUE THREAD

**Issue by richardeakin** [cinder/Cinder#996]
Carried over from #992.

Currently it is disabled as there is no `EventLog`, instead it is discussed that we should try `EventWrite` via COM.

**Comment by lucasvickers** [comment]
I did some exploration and committed [these changes](https://github.com/cinder/Cinder/compare/master...lucasvickers:log-eventwrite?expand=1) to my repo.

The two most obvious logging API candidates were [EventWrite](https://msdn.microsoft.com/en-us/library/windows/apps/aa363752.aspx) and [EventWriteString](https://msdn.microsoft.com/en-us/library/windows/apps/aa363750(v=vs.85).aspx).

### First EventWrite

The [EventWrite](https://msdn.microsoft.com/en-us/library/windows/apps/aa363752.aspx) API relies on a predefined logging event.  These events are defined by [EventDescriptors](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363754(v=vs.85).aspx).  If you take a look at the Remarks section of the EventDescriptors link, you'll see that we're supposed to use the Message Compiler to generate our manifest header.  This is different than the [prior](https://github.com/cinder/Cinder/blob/master/src/cinder/Log.cpp#556) way where we just defined the level and 0'd out the details at the time we were making the log call

In my example I keep getting ERROR_INVALID_PARAMETER (87), and I suspect it's because I have no valid EventDescriptors.

I tried digging a little deeper, and the issue is I have no app-specific provider.h file, which I believe is what we would generate when we register our provider (app?).  If you take a look at [this](https://msdn.microsoft.com/en-us/library/windows/apps/aa364162.aspx) document on writing manifest based events, it says to include ```provider.h```.  I tried this, but we seem to be picking up the default file and if you inspect ```provider.h``` it has no reference to message types.

### Then EventWriteString

The  [EventWriteString](https://msdn.microsoft.com/en-us/library/windows/apps/aa363750(v=vs.85).aspx) API seems far more promising (and similar to the old style of logging).  We don't need any event descriptors, etc.  The problem is that in ```evntprov.h``` ```EventWriteString``` is never defined due to the following check:
```
#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
```
A quick [search](http://stackoverflow.com/questions/15292199/when-does-if-winapi-family-partitionwinapi-partition-desktop-in-dxvahd-h-micr) tells me that we need to be using the desktop version of windows for this to be true, which probably runs counter to the whole WinRT thing.

### In Summary

It seems like we need to explore how to register our app as a provider (if that's the correct terminology anyway).
